uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/DatePicker.java

```diff
@@ -238,12 +250,75 @@ public void setForceParse( final boolean forceParse ) {
      * Set the format of the Date shown in the TextBox component.
      * This is converted to BS3's Date Format that the underlying jQuery-based BS3 DatePicker
      * uses to convert values in the TextBox to selections in the popup date picker element.
-     * @param gwtDateFormat
+     * @param dateFormat
      */
-    public void setFormat( final String gwtDateFormat ) {
-        this.gwtDateFormat = gwtDateFormat;
+    public void setFormat( final String dateFormat ) {
+        setFormat(LocaleInfo.getCurrentLocale().getLocaleName(),dateFormat);
+    }
+
+    public void setFormat(String currentLang,  final String dateFormat) {
+        this.gwtDateFormat = setDatePickerLangProperties(currentLang, dateFormat);
         this.gwtDateTimeFormat = DateTimeFormat.getFormat( this.gwtDateFormat );
-        datePicker.setFormat( DatePickerFormatUtilities.convertToBS3DateFormat( gwtDateFormat ) );
+        datePicker.setFormat(DatePickerFormatUtilities.convertToBS3DateFormat(this.gwtDateFormat));
+    }
+
+    /**
+     * Sets the org.gwtbootstrap3.extras.datepicker.client.ui.DatePicker language and week start day
+     * depending on the activeLocale passed as parameter.
+     * There are incompatibilities between DateTimeFormatInfoImpl and DatePickerResourceBundles for es and de locales.
+     * This function modify in that cases the dateFormat to avoid show the name of the months
+     * TODO review locales es and de showing the name of the month.
+     * @param activeLocale
+     * @param dateFormat
+     * @return the dateformat to be applied depending of locale. In case of es and de, avoiding show the name of month
+     */
+    private String setDatePickerLangProperties(String activeLocale, String dateFormat){
+        DatePickerLanguage datePickerLanguage = DatePickerLanguage.EN;
+        DatePickerDayOfWeek weekStart = DatePickerDayOfWeek.SUNDAY;
+        String format = dateFormat;
+        if (activeLocale.equals("es")) {
+            datePickerLanguage = DatePickerLanguage.ES;
```

Comment:
@nmirasch Glad it's of use.. 

I'd recommend double-checking at runtime (rather than relying on the values you use in your Unit Tests) as for example GWT's ```LocaleInfo``` JavaDocs state "Returns the name of this locale, such as 'default', 'en_US', etc". It's therefore feasible that 'en_US' won't match anything in ```DatePickerLanguage```. 

You might want to strip the language from the region (see https://github.com/droolsjbpm/kie-wb-distributions/blob/master/kie-wb-distribution-home/kie-wb-distribution-home-client/src/main/java/org/kie/workbench/common/screens/home/client/widgets/home/HomeImagesHelper.java#L63 as an example).